src/mixins/nr-affiliation-mixin.ts

```diff
@@ -194,101 +210,64 @@ export class NrAffiliationMixin extends Mixins(CommonMixin) {
   }
 
   /**
-   * Handle "Incorporate Now" button.
+   * Handle the action buttons (numbered selection).
```

Comment:
I took the liberty of refactoring all the following blocks of code here.

Previously, we had 2 functions for each entity type. It was OK at first when we had only incorporation and registration. But the more stuff we add, the more these functions are increasing.

We went from 2 to 4 when we added amalgamations. And If I added here continuation in, it'll be 6 functions doing something similar. And that number will continue on increasing. It's a pain adding more stuff.

So, I refactored to have only 2 functions that deal with everything. Now, adding more types/filings is by simply adding a conditional and what that type is going to have.

Reply:
I totally agree with you 👍